service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -268,6 +264,35 @@ public boolean updateWorkspace(
     return updated;
   }
 
+  /** Update a workspace properties */
+  @WriteTransaction
+  public boolean updateWorkspaceProperties(UUID workspaceUuid, Map<String, String> propertyMap) {
+    // get current property in this workspace id
+    String selectPropertiesSql = "SELECT properties FROM workspace WHERE workspace_id = :id";
+    MapSqlParameterSource propertiesParams =
+        new MapSqlParameterSource().addValue("id", workspaceUuid.toString());
+    String result =
+        jdbcTemplate.queryForObject(selectPropertiesSql, propertiesParams, String.class);
+    logger.info("Retrieved workspace record {}", result);
+    Map<String, String> properties = DbSerDes.jsonToProperties(result);
+    properties.putAll(propertyMap);
+    final String sql =
+        "UPDATE workspace SET properties = cast(:properties AS jsonb) WHERE workspace_id = :id";
+
+    var params = new MapSqlParameterSource();
+    params
+        .addValue("properties", DbSerDes.propertiesToJson(properties))
+        .addValue("id", workspaceUuid.toString());
+
+    int rowsAffected = jdbcTemplate.update(sql, params);
+    boolean updated = rowsAffected > 0;
```

Comment:
The `queryForObject` will throw if the workspace record is not found, so you will never get here if it doesn't exist. So the update will always return 1. I think the rest of this logic is not needed. In fact, do we even need or use the return value?

Reply:
that's a good point Dan..  We need the return value to log a change activity if an update happened. 

There are two ways to do this:
1. we compare the hashmap before and after putAll to see if the value does change
2. We assume that the properties is changed if we reached the end. Don't return anything here and just log an update in WorkspaceService.

I'm OK with just doing 2.

Reply:
Thank you Yu. I am ok with both ways. I think once the updateProperties method was called, must have something update even if update a=b to a=b. 
I am wondering, at this moment, I added the try catch like all other methods. Does this still need to delete the return? 
If the update method should delete the return. just double check, should I follow this in Delete method? (I mean PF-1792)

Reply:
I'd make this method `void` and removing `rowsAffected` and related processing.
If the transaction succeeds, then the `WorkspaceService.updateWorkspaceProperties` can proceed with writing the activity. If it fails, then it will have thrown an exception that will be handled by the global exception handler.

Reply:
Done. Dan. Thanks for the explanation. I just correct it. Let me know if there any more ideas. 
really appreciate for the last minute before your vacation.